mlkem/src/native/riscv64/src/rv64v_poly.c

```diff
@@ -0,0 +1,750 @@
+/*
+ * Copyright (c) The mlkem-native project authors
+ * SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
+ */
+
+/*  === Kyber NTT using RISC-V Vector intrinstics */
+
+#include "../../../common.h"
+
+#if defined(MLK_ARITH_BACKEND_RISCV64) && \
+    !defined(MLK_CONFIG_MULTILEVEL_NO_SHARED)
+
+#include <riscv_vector.h>
+#include "arith_native_riscv64.h"
+#include "rv64v_settings.h"
+
+static inline vint16m1_t fq_redc(vint16m1_t rh, vint16m1_t rl, size_t vl)
+{
+  vint16m1_t t;
+  vbool16_t c;
+
+  t = __riscv_vmul_vx_i16m1(rl, MLK_RVV_QI, vl); /* t = l * -Q^-1  */
+  t = __riscv_vmulh_vx_i16m1(t, MLKEM_Q, vl);    /* t = (t*Q) / R  */
+  c = __riscv_vmsne_vx_i16m1_b16(rl, 0, vl);     /* c = l == 0     */
```

Comment:
The comment looks wrong here. Should it be `c = (l != 0)`?

Why are we not using a subtract on the high parts instead?

Reply:
Fixed the comment, but the other question still stands.